tools/chpldoc/findUndocumentedSymbols.py

```diff
@@ -0,0 +1,387 @@
+            and is_docstring_comment(prev)
+        ):
+            return True
+
+        return False
+
+    def __call__(self):
+        syms = [s for s in self.get_documentable_symbols([a for a in self.ast])]
+        for s in syms:
+            if not is_nodoc(s) and not self.has_doc_comment(s):
+                if self.ignore_deprecated and is_deprecated(s):
+                    continue
+                if self.ignore_unstable and is_unstable(s):
+                    continue
+                yield s
+
+
+def get_files(files: List[str], check_suffix=False) -> Generator:
+    """
+    Yield all chapel files, following directories recursively
+    """
+
+    for file in files:
+        if os.path.isfile(file):
+            # just yield any file given if not checking the suffix
+            if not check_suffix:
+                yield file
+            elif check_suffix and os.path.splitext(file)[1] == ".chpl":
+                yield file
+
+        elif os.path.isdir(file):
+            for root, subdirs, files in os.walk(file):
+                # yield files in directory files, always check the suffix
+                yield from get_files([os.path.join(root, file) for file in files], True)
+                # recurse into sub dirs, always checl the suffix
+                yield from get_files([os.path.join(root, d) for d in subdirs], True)
+        else:
+            print("Error: path was not file or directory", file=sys.stderr)
+            exit(1)
+
+
+def get_trees(filenames: List[str]) -> Generator:
+    """
+    Yield dyno asts and their filename.
+    Some files might have the same name, which Dyno really doesn't like.
+    Strateify files into "buckets"; within each bucket, all filenames are
+    unique. Between each bucket, re-create the Dyno context to avoid giving
+    it complicting files.
+    """
+    basenames = defaultdict(lambda: 0)
+    buckets = defaultdict(lambda: [])
+    for filename in filenames:
+        filename_expanded = os.path.realpath(os.path.expandvars(filename))
+
+        basename = os.path.basename(filename_expanded)
+        bucket = basenames[basename]
+        basenames[basename] += 1
+        buckets[bucket].append((filename, filename_expanded))
+
+    for filenames in buckets.values():
+        ctx = dyno.Context()
+        for filename, filename_expanded in filenames:
+            ast = ctx.parse(filename_expanded)
+            yield (filename, ast)
+
+
+def main(raw_args: List[str]) -> int:
+    a = ap.ArgumentParser()
+    a.add_argument("files", nargs="*")
+    a.add_argument("--ignore-deprecated", action="store_true", default=False)
+    a.add_argument("--ignore-unstable", action="store_true", default=False)
+    args = a.parse_args(raw_args)
+
+    for filename, astList in get_trees(get_files(args.files)):
+        fus = FindUndocumentedSymbols(
+            astList,
+            ignore_deprecated=args.ignore_deprecated,
+            ignore_unstable=args.ignore_unstable,
```

Comment:
I wonder if just doing `**args` would work here. I don't think it will, but wouldn't it be nice?